<!-- -*- mode: XML -*- -->
<launch>
    <arg name="connect_on_startup"   default="true"/>
    <arg name="calibrate_on_startup" default="true"/>
    <arg name="engage_on_startup"    default="true"/>

    <arg name="publish_odom"        default="true"/>
    <arg name="publish_odom_tf"     default="true"/>
    
    <arg name="output"    default="screen"/>
    <arg name="respawn"   default="false"/>

    <arg name="twist_control"   default="true"/>
    <arg name="joint_control"   default="true"/>

    <arg name="use_preroll"   default="true"/>

    <arg name="rpm_gain"   default="1.5"/>
    <arg name="axis_for_right"   default="1"/>

    <arg name="front_serial_number"   default="20803592524B"/>
    <arg name="back_serial_number"    default="20583592524B"/>
    
<group ns="front">
    <node pkg="odrive_ros" type="odrive_node" name="odrive" ns="odrive" 
            output="$(arg output)" respawn="$(arg respawn)">
        <param name="wheel_track" value="0.285"/>
        <param name="tyre_circumference" value="0.341"/>
        
        <param name="connect_on_startup"   value="$(arg connect_on_startup)"/>
        <param name="calibrate_on_startup" value="$(arg calibrate_on_startup)"/>
        <param name="engage_on_startup"    value="$(arg engage_on_startup)"/>
        
        <param name="publish_odom"         value="$(arg publish_odom)"/>
        <param name="publish_odom_tf"      value="$(arg publish_odom_tf)"/>

        <param name="axis_for_right"      value="$(arg axis_for_right)"/>

        <param name="twist_control"        value="$(arg joint_control)"/>
        <param name="joint_control"        value="$(arg twist_control)"/>

        <param name="use_preroll"          value="$(arg use_preroll)"/>
        <param name="rpm_gain"          value="$(arg rpm_gain)"/>

        <param name="serial_number"  type="str" value="$(arg front_serial_number)"/>
    </node>
</group>
<group ns="back">
    <node pkg="odrive_ros" type="odrive_node" name="odrive" ns="odrive" 
            output="$(arg output)" respawn="$(arg respawn)">
        <param name="wheel_track" value="0.285"/>
        <param name="tyre_circumference" value="0.341"/>
        
        <param name="connect_on_startup"   value="$(arg connect_on_startup)"/>
        <param name="calibrate_on_startup" value="$(arg calibrate_on_startup)"/>
        <param name="engage_on_startup"    value="$(arg engage_on_startup)"/>
        
        <param name="publish_odom"         value="$(arg publish_odom)"/>
        <param name="publish_odom_tf"      value="$(arg publish_odom_tf)"/>

        <param name="twist_control"        value="$(arg joint_control)"/>
        <param name="joint_control"        value="$(arg twist_control)"/>

        <param name="use_preroll"          value="$(arg use_preroll)"/>
        <param name="rpm_gain"          value="$(arg rpm_gain)"/>

        <param name="serial_number"  type="str" value="$(arg back_serial_number)"/>
    </node>
</group>
</launch>
